server/lib/reverse_etl/utils/cursor_query_builder.rb

```diff
@@ -6,21 +6,38 @@ class CursorQueryBuilder
       def self.build_cursor_query(sync_config, current_cursor_field)
         existing_query = sync_config.model.query
         query_type = sync_config.source.query_type || "raw_sql"
-        if current_cursor_field
-          cursor_condition = case query_type.to_sym
-                             when :soql
-                               "#{sync_config.cursor_field} >= #{current_cursor_field}"
-                             when :raw_sql
-                               "#{sync_config.cursor_field} >= '#{current_cursor_field}'"
-                             end
+
+        cursor_condition = build_cursor_condition(sync_config.cursor_field, current_cursor_field, query_type)
+        case query_type.to_sym
+        when :soql
+          if cursor_condition.present?
+            where_clause = existing_query.include?("WHERE") ? " AND #{cursor_condition}" : " WHERE #{cursor_condition}"
+            "#{existing_query}#{where_clause} ORDER BY #{sync_config.cursor_field} ASC"
+          else
+            "#{existing_query} ORDER BY #{sync_config.cursor_field} ASC"
+          end
+        when :raw_sql
+          if cursor_condition.present?
+            "#{existing_query} AS subquery " \
+            "WHERE #{cursor_condition} " \
+            "ORDER BY #{sync_config.cursor_field} ASC"
+          else
+            "#{existing_query} AS subquery " \
+            "ORDER BY #{sync_config.cursor_field} ASC"
```

Comment:
Why do we need cursor field if the cursor condition is not present ?

Reply:
during the initial fetch, we need to get records in ascending order of the cursor field and save the value of the last record's cursor field in the current_cursor_field variable